pkg/util/admissioncheck/admissioncheck.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissioncheck
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+var (
+	ErrNilParametersRef = errors.New("missing parameters reference")
+	ErrBadParametersRef = errors.New("bad parameters reference")
+)
+
+type objAsPtr[T any] interface {
+	client.Object
+	*T
+}
+
+type ConfigHelper[PtrConfT objAsPtr[ConfT], ConfT any] struct {
+	gk     schema.GroupKind
+	client client.Client
+}
+
+func NewConfigHelper[PtrConfT objAsPtr[ConfT], ConfT any](c client.Client) (*ConfigHelper[PtrConfT, ConfT], error) {
+	helper := ConfigHelper[PtrConfT, ConfT]{
+		client: c,
+	}
+
+	gvk, err := apiutil.GVKForObject(helper.newConfigPtr(), c.Scheme())
```

Comment:
Can we eliminate retrieval of the GK from object? It would eliminate the need for returning error, and going via interim object.

Reply:
I wold like to keep it as is.

Reply:
I think this comment is non-blocking to me. For visibility @alculquicondor.